python/ray/util/sgd/torch/worker_group.py

```diff
@@ -107,6 +107,23 @@ def validate(self, num_steps=None, profile=False, info=None):
         raise NotImplementedError
 
 
+def wait_for_condition(condition_predictor, timeout=60, retry_interval_ms=100):
+    """Wait until a condition is met or time out with an exception.
+    Args:
+        condition_predictor: A function that predicts the condition.
+        timeout: Maximum timeout in seconds.
+        retry_interval_ms: Retry interval in milliseconds.
+    Raises:
+        RuntimeError: If the condition is not met before the timeout expires.
+    """
+    start = time.time()
+    while time.time() - start <= timeout:
+        if condition_predictor():
+            return
+        time.sleep(retry_interval_ms / 1000.0)
+    raise RuntimeError("The condition wasn't met before the timeout expired.")
```

Comment:
Instead of raising an Error here can we just have this return True or False? Then inside the `worker_group` we can take the appropriate action based on the output.

Reply:
I have refactored according to comment.